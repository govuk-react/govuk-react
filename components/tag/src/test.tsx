import * as React from 'react';
import { mount } from 'enzyme';

import { Tag } from '.';

describe('Tag', () => {
  it('can render an inactive tag', () => {
    mount(<Tag inactive>inactive</Tag>);
  });

  it('can render an tinted tag', () => {
    mount(<Tag tint="GREY">inactive</Tag>);
  });

  it('matches wrapper snapshot', () => {
    expect(mount(<Tag>beta</Tag>)).toMatchSnapshot('wrapper mount');
  });

  it('matches snapshot with custom colour', () => {
    expect(
      mount(
        <Tag backgroundColor="#0b0c0c" color="#d53880">
          beta
        </Tag>
      )
    ).toMatchSnapshot('with custom colour');
  });
});
