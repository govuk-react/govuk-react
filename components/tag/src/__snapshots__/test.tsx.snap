// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag matches snapshot with custom colour: with custom colour 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 14px;
  line-height: 1;
  display: inline-block;
  padding-top: 5px;
  padding-right: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  outline: 2px solid transparent;
  outline-offset: -2px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  background-color: #1d70b8;
  color: #ffffff;
  background-color: #0b0c0c;
  color: #d53880;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 16px;
    line-height: 1;
  }
}

<Tag
  backgroundColor="#0b0c0c"
  color="#d53880"
  inactive={false}
  tint="SOLID"
>
  <strong
    className="c0"
    color="#d53880"
  >
    beta
  </strong>
</Tag>
`;

exports[`Tag matches wrapper snapshot: wrapper mount 1`] = `
.c0 {
  font-family: "nta",Arial,sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  font-weight: 700;
  font-size: 14px;
  line-height: 1;
  display: inline-block;
  padding-top: 5px;
  padding-right: 8px;
  padding-bottom: 4px;
  padding-left: 8px;
  outline: 2px solid transparent;
  outline-offset: -2px;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  background-color: #1d70b8;
  color: #ffffff;
}

@media print {
  .c0 {
    font-size: 14px;
    line-height: 1;
  }
}

@media only screen and (min-width:641px) {
  .c0 {
    font-size: 16px;
    line-height: 1;
  }
}

<Tag
  inactive={false}
  tint="SOLID"
>
  <strong
    className="c0"
  >
    beta
  </strong>
</Tag>
`;
